src/syscalls/interfaces/uhyve.rs

```diff
@@ -218,39 +218,44 @@ impl SyscallInterface for Uhyve {
 	#[cfg(target_os = "hermit")]
 	fn get_application_parameters(&self) -> (i32, *const *const u8, *const *const u8) {
 		// determine the number of arguments and environment variables
-		let mut syscmdsize = SysCmdsize::new();
-		uhyve_send(UHYVE_PORT_CMDSIZE, &mut syscmdsize);
+		let syscmdsize = SysCmdsize::new();
+		hypercall(UHYVE_PORT_CMDSIZE, &syscmdsize);
+
+		let argc = unsafe { ptr::read_unaligned(&syscmdsize.argc) as usize };
+		let envc = unsafe { ptr::read_unaligned(&syscmdsize.envc) as usize };
```

Comment:
I know I have been nagging about the `&mut`, but I am not sure if I like this solution either. Can you move the allocation of the memory inside `hypercall` and just return it from the call?

Reply:
Hm, in this case you create a function for every possible hypercall. Do think this is more understandable?

Reply:
How about returning an enum that fits all needs?